File: abc/abc166/abc166_d.cs
// https://atcoder.jp/contests/abc166/submissions/12786804

using System;

class Program
{
    static void Main()
    {
        var X = int.Parse(Console.ReadLine());

        static int Pow5(int x) => x * x * x * x * x;
        for (int i = -118; i <= 119; i++)
        {
            for (int j = -119; j <= 118; j++)
            {
                if (Pow5(i) - Pow5(j) == X)
                {
                    Console.WriteLine($"{i} {j}");
                    return;
                }
            }
        }
    }
}

File: abc/abc166/abc166_d.fs
// https://atcoder.jp/contests/abc166/submissions/12789378

let X = stdin.ReadLine() |> int

let s = 120

let pow5 x = x * x * x * x * x

let rec solver i j =
    if j = s then solver (i + 1) -s
    elif pow5 i - pow5 j = X then
        i.ToString() + " " + j.ToString() |> stdout.WriteLine
        ()
    else solver i (j + 1)

solver -s -s
